ddgr returns 0 exit code on HTTP Errors

When using dggr with -w scholar.google.com ,  google may after a while, block the request.  In this scenario, "[ERROR] HTTP Error 403: Forbidden" is sent to stderr, then ddgr exits, and the exit code is 0.

Internally, the fetch routines raise a DGConnectionError,  calls LOGERR(e), and returns None, after which dggr exits with no error code.

This makes it difficulty to use ddgr from another script, for example to loop over a set of searches and wait and repeat the last search, when google blocks.

As a temporary measure,  I inserted a line  sys.exit(1)  after the call to LOGERR(e), perhaps depending on the content of the error.

--- a/ddgr
+++ b/ddgr
@@ -838,8 +838,7 @@ class DdgConnection:
                                   expected_code=200)
         except Exception as e:
             LOGERR(e)
-            return None
-
+            sys.exit(1)
         return r
 
 

